### DISCUSSION TITLE
Wallbox-Steuerung analog zum senec-charger?

### DISCUSSION THREAD

**Discussion by gereons** [solectrus/solectrus#3223] [Ideas]
Bei tibber hat heute zeitweise wegen eines Fehlers der Strombörse [eine kWh etwa 3 Euro gekostet](https://www.elektroauto-news.net/news/tibber-strompreis-kilowattstunde-drei-euro).

Da ich gerade mit dem Gedanken spiele Anfang 2025 zu tibber zu wechseln, und hohen Stromverbrauch in solchen Situationen natürlich gern vermeiden würde, folgende Idee:

Solectrus kennt ja über den `tibber-collector` schon die anstehenden Preise, und der `senec-charger` kann mit dieser Info den Akku im Keller laden.

Wäre es nicht sinnvoll, diesen Mechanismus so zu erweitern, dass ein analoger  `car-charger` das Auto (nachts) lädt, wenn's gerade am preiswertesten ist? Das würde natürlich eine Ansteuerung der Wallbox benötigen, aber zumindestens für Senec-Wallboxen die per App gesteuert werden können wäre das dem `senec`-Gem vermutlich sehr einfach beizubringen, da es einen wichtigen Teil (das Login) der Cloud-API ja schon beherrscht - Status und Kontrolle der Wallbox geschieht über einen ziemlich [simplen Endpunkt](https://documenter.getpostman.com/view/10329335/UVCB9ihW#25cbd682-7ea2-41e9-ad86-df56697d8736).

(Diesen Endpunkt benutze ich schon seit geraumer Zeit für eine ziemlich primitive Steuerung via cronjobs mit bash, curl und [sunwait](https://github.com/risacher/sunwait), aber richtig automatisiert wäre natürlich schöner :-) )

**Comment by ledermann** [comment]
Interessante Idee, über die ich erst mal länger nachdenken musste ;-)

Wenn ich es richtig verstehe, ist sowas bereits jetzt mit evcc möglich:
https://github.com/evcc-io/evcc/discussions/4447

Ich bin mir nicht sicher, ob eine eigene Implementierung, die noch dazu nur für SENEC funktionieren würde, ökonomisch sinnvoll wäre. Den SENEC-Charger habe ich nur gebaut, weil das über ein Sponsoring finanziert wurde.

**Comment by gereons** [comment]
Von evcc habe ich bisher Abstand genommen, weil das nach meinem Verständnis die alleinige Kontrolle über die Wallbox übernehmen will und dazu einen direkten Modbus-Anschluss an die WB braucht. Der wiederum würde bei mir zu einem ziemlichen Umbau der Hardware führen, den ich mir gerne ersparen würde. Ganz allgemein ist es mir auch deutlich sympathischer wenn es nicht mehrere (potenziell konkurrierende) Systeme gibt, die gleichzeitig 'was in meinem Haus steuern/monitoren wollen.

Was die Ökonomie (ich nehme an die der Softwareentwicklung und -wartung ist gemeint) angeht: das hängt natürlich stark davon ab, was die Userschaft so installiert hat. Wenn nur ein paar Prozent so eine komplett-alles-von-Senec-Installation haben ist der Markt wahrscheinlich zu klein um den Aufwand zu treiben, klar. Wenn so ein Setup aber eher Standard ist könnte ich mir so ein Feature gut als kostenpflichtige Erweiterung des Basis-Funktionsumfangs vorstellen - mich hättest du als Kunde :-)

**Reply by ledermann** [reply]
Wenn ich es richtig sehe, könnte das Feature nützlich sein für SENEC-User mit SENEC-Wallbox und Tibber-Vertrag. Also eine vermutlich eher kleine Zielgruppe ;-) Eine spätere Anpassbarkeit an andere Wallboxen sollte man also im Blick behalten.

Bei der Logik müsste man sich etwas überlegen. Als Input-Parameter würde ich momentan sehen:

- Gewünschter Mindestladestand des Autos bis zum Zeitpunkt X ("morgen um 07:00 fahre ich zum Kundentermin und muss dann mindestens 50% im Akku haben")
- Gewünschter Höchstladestand des Autos ("zur Akku-Schonung nicht mehr als 80% laden")
- Größe des Auto-Akkus und Ladegeschwindigkeit der Wallbox (denn davon hängt es ab, wie lange die Beladung dauern wird, wichtig zur Ermittlung des optimalen Zeitfensters)
- Jeweils aktueller Ladestand des Autos (siehe #2881)

Da die Parameter zum gewünschte Ladestand sich täglich ändern können, müssten diese sinnvollerweise über das UI (von SOLECTRUS) erreichbar sein, was die bisherige Architektur etwas über den Haufen werfen würde. Die App ist bisher auf read-only festgelegt, sie müsste dafür aber andere Container/Services benachrichtigen können.

Ja, als kostenpflichtiges Zusatzfeature könnte ich mir das vorstellen. Kurzfristig wird das zwar nicht klappen, da ich erst ein paar andere Dinge bei SOLECTRUS auf die Spur bringen möchte, aber die Idee ist auf jeden Fall spannend 👍

**Reply by gereons** [reply]
Ja genau, das wäre die Ideallösung. Danke für's Feedback.

Ein Punkt bzgl Zielgruppe noch: ein Feature "sorge dafür das mein Auto an diesen Tagen morgens geladen ist" ist sicher auch für Leute ohne Tibber nützlich.
